### PR TITLE
Adding serialize function to boundingbox class

### PR DESCRIPTION
**Pull request by GivAlz**
As suggested by @bangerth I'm splitting https://github.com/dealii/dealii/pull/5523 into two PR. This is the easy one, adding a serialize function to bounding box. I shall need a bit longer for the other PR...

Best,
Giovanni

### CODE REVIEW

File: include/deal.II/base/bounding_box.h [hunk 1]
```diff
@@ -134,8 +136,17 @@ class BoundingBox
    */
   double volume() const;
 
+  /**
+   * Boost serialization function
+   */
+  template<class Archive>
+  void serialize(Archive &ar, const unsigned int /*version*/)
```
Comment (davydden) [inline include/deal.II/base/bounding_box.h]: please separate definition and declaration.

Comment (luca-heltai) [reply include/deal.II/base/bounding_box.h]: (you can still put the declaration in the h file, at the end, enclosed in `#ifndef DOXYGEN ... #endif`).

Comment (davydden) [reply include/deal.II/base/bounding_box.h]: not only you can, but i think you should as it's templated with `Archive` and we can not instantiate it for some a priori known types.

File: tests/base/bounding_box_4.cc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// testing serialize function for class boundingbox
```
Comment (bangerth) [inline tests/base/bounding_box_4.cc]: -> BoundingBox

**Comment by bangerth** [comment]
Thanks, @GivAlz !

Please make the two changes requested and this should be good to go!

**Comment by GivAlz** [comment]
You're welcome, @bangerth ... and thank you for all the suggestions :)

Ok, pushed everything into a single commit, now to another PR ...

File: include/deal.II/base/bounding_box.h [hunk 3]
```diff
@@ -134,8 +136,14 @@ class BoundingBox
    */
   double volume() const;
 
+  /**
+   * Boost serialization function
+   */
+  template<class Archive>
+  void serialize(Archive &ar, const unsigned int /*version*/);
```
Comment (davydden) [inline include/deal.II/base/bounding_box.h]: you can keep `const unsigned int version` here.

**Comment by GivAlz** [comment]
Ok, I removed the unnecessary line, now everything should be fine :)

**Comment by bangerth** [comment]
/run-tests
OK to merge once the tester is happy.

**Pull request merged**